### PR TITLE
AM2R: add custom darkness/submerged chaos options

### LINKED ISSUE #5200: AM2R: Option to have random dark areas
Similar to how prime does it. for a crude experimental option, it may be sufficient to just set the darkness level in CC
But depending on how this'll be implemented on the RDV side, it may be required to go through the whole DB, and set darkness extra data for every room and feed that also into the patcher.

### LINKED ISSUE #6705: Add option to have random chance of a room being completely flooded
Would be a funny chaos option

### PR DESCRIPTION
**Pull request by Miepee**
fixes #6705 
fixes #5200 

image

### CODE REVIEW

**Comment by Miepee** [comment]
This is ready for review now.

File: randovania/games/am2r/exporter/patch_data_factory.py [hunk 1]
```diff
@@ -177,15 +177,41 @@ def _create_pickups_dict(
...
+                        "liquid_type": liquid_type,
+                        "liquid_level": -100,
+                        "should_be_at_very_front": True,
+                    }
+
+                return_dict[area.extra["map_name"]] = {
+                    "display_name": area.name,
+                    "region_name": region.name,
+                    "minimap_data": area.extra["minimap_data"],
+                    "light_level": str(light_level),
```
Comment (henriquegemignani) [inline randovania/games/am2r/exporter/patch_data_factory.py:210]: The string "None" is applicable here?

File: randovania/games/am2r/gui/ui_files/preset_am2r_chaos.ui [hunk 2]
```diff
@@ -0,0 +1,347 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (henriquegemignani) [inline randovania/games/am2r/gui/ui_files/preset_am2r_chaos.ui:1]: Screenshot please?

Comment (Miepee) [reply randovania/games/am2r/gui/ui_files/preset_am2r_chaos.ui:1]: In PR description

File: randovania/games/am2r/layout/am2r_configuration.py [hunk 3]
```diff
@@ -58,4 +66,16 @@ def unsupported_features(self) -> list[str]:
         if self.artifacts.required_artifacts > self.artifacts.placed_artifacts:
             result.append("The amount of required DNA cannot be higher than the total amount of placed DNA.")
 
+        if self.darkness_min > self.darkness_max:
+            result.append("The minimum darkness value cannot be higher than the maximum darkness value.")
```
Comment (henriquegemignani) [inline randovania/games/am2r/layout/am2r_configuration.py:70]: Add some test for these cases please

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

File: randovania/games/am2r/exporter/patch_data_factory.py (hunk 1 above)
Comment (Miepee) [reply randovania/games/am2r/exporter/patch_data_factory.py:210]: whoops, no. 
Fixed locally.

File: randovania/games/am2r/layout/am2r_configuration.py (hunk 3 above)
Comment (Miepee) [reply randovania/games/am2r/layout/am2r_configuration.py:70]: added
Also removed the second case here, cause it felt unnecessary.

File: randovania/games/am2r/exporter/patch_data_factory.py [hunk 4]
```diff
@@ -177,15 +177,41 @@ def _create_pickups_dict(
...
-            for region in self.game.region_list.regions
-            for area in region.areas
-        }
+        rng = Random(self.description.get_seed_for_player(self.players_config.player_index))
+
+        return_dict = {}
+        for region in self.game.region_list.regions:
+            for area in region.areas:
+                light_level = None
+                if rng.random() < (self.configuration.darkness_chance / 1000.0):
```
Comment (henriquegemignani) [inline randovania/games/am2r/exporter/patch_data_factory.py:186]: I feel 1000 units is more precision than we need and inflates the permalink unecessarily.
But I also feel we're past the point we care much about permalink size 😅

**Review by henriquegemignani — APPROVED** [review]

File: randovania/games/am2r/exporter/patch_data_factory.py (hunk 4 above)
Comment (Miepee) [reply randovania/games/am2r/exporter/patch_data_factory.py:186]: Could've used a float, but I dont think it really matters in terms of permalink size.

**Pull request merged**